crypto/s2n_ecdsa.c

```diff
@@ -46,29 +46,43 @@ S2N_RESULT s2n_ecdsa_der_signature_size(const struct s2n_pkey *pkey, uint32_t *s
     return S2N_RESULT_OK;
 }
 
-static int s2n_ecdsa_sign(const struct s2n_pkey *priv, s2n_signature_algorithm sig_alg,
-        struct s2n_hash_state *digest, struct s2n_blob *signature)
+int s2n_ecdsa_sign_digest(const struct s2n_pkey *priv, struct s2n_blob *digest, struct s2n_blob *signature)
 {
-    sig_alg_check(sig_alg, S2N_SIGNATURE_ECDSA);
+    POSIX_ENSURE_REF(priv);
+    POSIX_ENSURE_REF(digest);
+    POSIX_ENSURE_REF(signature);
 
     const s2n_ecdsa_private_key *key = &priv->key.ecdsa_key;
     POSIX_ENSURE_REF(key->ec_key);
 
-    uint8_t digest_length;
+    unsigned int signature_size = signature->size;
+    POSIX_GUARD_OSSL(ECDSA_sign(0, digest->data, digest->size, signature->data, &signature_size, key->ec_key), S2N_ERR_SIGN);
```

Comment:
Is there a name you can give the first argument? Apparently it's `type` so I would assume it would have a define for it.